src/plugin/ipc/file/fileconnlist.cpp

```diff
@@ -490,13 +490,22 @@ FileConnList::recreateShmFileAndMap(const ProcMapsArea &area)
     JASSERT(fd != -1) (JASSERT_ERRNO);
   }
 
-  // Get to the correct offset.
-  JASSERT(lseek(fd, area.offset, SEEK_SET) == area.offset) (JASSERT_ERRNO);
-
-  // Now populate file contents from memory.
-  JASSERT(Util::writeAll(fd, area.addr, area.size) == (ssize_t)area.size)
-    (JASSERT_ERRNO);
+  // Unlink (the area was originally unlinked)
+  unlink(area.name);
+
+  // Restore SHM area mapping
+  // The mmap in restoreShmArea will clear the content in the area because the
+  // file has been unlinked, so we have to save the content first into a
+  // temp buffer
+  void* tempbuf = mmap(NULL, area.size,
+                       PROT_READ|PROT_WRITE, MAP_PRIVATE|MAP_ANONYMOUS, 0, 0);
+  JASSERT(tempbuf > (void*)0) (MAP_FAILED);
```

Comment:
Please change it to `JASSERT(tempbuf != MAP_FAILED) (JASSERT_ERRNO) (area.size);`